src/app/parser/open-api/param-types.ts

```diff
@@ -275,5 +283,203 @@ export function getEmptySchema(): Schema {
     required: undefined,
     description: undefined,
     _$rendered: undefined,
+    _$requiresRelaxedTypeTag: undefined,
   };
 }
+
+/**
+ * Function to determine if this scalar type schema requires
+ * relaxed type tag
+ *
+ * More Reading: https://github.com/hasura/ndc-nodejs-lambda?tab=readme-ov-file#relaxed-types
+ *
+ * @param schema
+ */
+export function isRelaxedTypeTagRequiredForScalarTypeSchema(
+  schema: SchemaTypeScalar,
+): boolean {
+  return (
+    (scalarSchemaIsObject(schema) || (schema.enum && schema.enum.length > 1)) ??
+    false
+  );
+}
+
+/**
+ * Function to determine if this object type schema requires
+ * relaxed type tag
+ *
+ * More Reading: https://github.com/hasura/ndc-nodejs-lambda?tab=readme-ov-file#relaxed-types
+ *
+ * ### WARNING
+ * This function should be called after all the children/properties of the schema
+ * have been processed. This function *DOES NOT* process the tag requirements for the
+ * schema's children/properties
```

Comment:
QQ: Can't we call this function while processing the properties of the schema?

Reply:
That's a very good question, and it is something I wanted to do too to avoid the possibility of human-errors. But, it cannot be done. Or, to put it another way, I couldn't figure out a way to do it while processing the properties, because while traversing a schema can be standardized, the operation that should happen once that traversing is done will be unique to the use case (for example, object schema being rendered here: [renderObjectTypeSchema](https://github.com/hasura/ndc-open-api-lambda/blob/e6a4941aaa3a473d3a4f57942911579ae2a185c0/src/app/generator/param-generator.ts#L115)). Adding additional traversal logic for properties that have already been processed just increases the execution time of the program, and it's unnecessary because the processing has already happened